proposals/0000-keyboard-apis.md

```diff
@@ -0,0 +1,142 @@
+      
+  private _onKeyDown = (event: IKeyboardEvent) => {
+    this.setState({ lastKeyDown: event.nativeEvent.key });
+  };
+  
+```
+
+### Example 2 : Custom key stroke handling in native components
+
+In the following example, the app's logic takes precedence when certain keystrokes are encountered in the TextInput before the native platform can handle them. 
+
+```
+  <TextInput onKeyUp={this._onKeyUp} keyUpEvents={handledNativeKeyboardEvents} />
+  
+  const handledNativeKeyboardEvents: IHandledKeyboardEvent[] = [
+     { key: 'Enter' },
+     { key: 'Esc' },
+  ];
+  
+  private _onKeyUp = (event: IKeyboardEvent) => {
+    if(event.nativeEvent.key == 'Enter' && event.nativeEvent.eventPhase == KeyEventPhase.AtTarget){
+            //do something custom when Enter key is detected when focus is on the TextInput component
+    } else if (event.nativeEvent.key == 'Esc' && event.nativeEvent.eventPhase == KeyEventPhase.AtTarget){
+            //do something custom when Escape key is detected when focus is on the TextInput component
+    }    
+  };
+  
+```
+
+## Detailed design
+
+The APIs being introduced here will follow the models created by :
+- The corresponding React APIs for [Keyboarding events](https://reactjs.org/docs/events.html#keyboard-events)
+- The corresponding Windows APIs for [Keyboarding events](https://docs.microsoft.com/en-us/windows/uwp/design/input/keyboard-events)
+
+### Keyboard events
+The following events will be introduced on `View` and `TextInput` to cover the most common use cases where key stroke handling is likely to occur. Other individual components where they may be neeeded can wrap a `View` around to capture the Key events.
+
+| API | Args | Returns | Description |
+|:---:|:----:|:-------:|:----:|
+| onKeyDown | IKeyboardEvent | void | Occurs when a keyboard key is pressed when a component has focus. On Windows, this corresponds to [KeyDown](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.keydown)|
+| onKeyDownCapture | IKeyboardEvent | void | Occurs when the `onKeyDown` event is being routed. `onKeyDown` is the corresponding bubbling event. On Windows, this corresponds to [PreviewKeyDown](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.previewkeydown)|
+| onKeyUp | IKeyboardEvent | void | Occurs when a keyboard key is released when a component has focus. On Windows, this corresponds to [KeyUp](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.keyup)|
+| onKeyUpCapture | IKeyboardEvent | void | Occurs when the `onKeyUp` event is being routed. `onKeyUp` is the corresponding bubbling event. On Windows, this corresponds to [PreviewKeyUp](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.previewkeyup)|
+
+Where `IKeyboardEvent` will be a new event type added to `ReactNative.NativeSyntheticEvents` of type `INativeKeyboardEvent`. `INativeKeyboardEvent` is a new interface and will expose the following properties:
```

Comment:
Does this describe the properties that will be passed to the javascript layer or is this specifically for the native interface?

If this implies the javascript event object as well it seems like there are some things missing from the react synthetic event, such as charCode

Reply:
This is for the JS layer. The charCode might be interesting if we were to add onKeyPress in addition to KeyDown/KeyUP. For now, we are not adding KeyPress - we can discuss this during the review to see if there are scenarios impacted right away that need it.